analyzers/src/SonarAnalyzer.VisualBasic/Rules/SecurityPInvokeMethodShouldNotBeCalled.cs

```diff
@@ -0,0 +1,64 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.VisualBasic;
+using Microsoft.CodeAnalysis.VisualBasic.Syntax;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules.VisualBasic
+{
+    [DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+    [Rule(DiagnosticId)]
+    public sealed class SecurityPInvokeMethodShouldNotBeCalled : SecurityPInvokeMethodShouldNotBeCalledBase<SyntaxKind, InvocationExpressionSyntax>
+    {
+        protected override SyntaxKind SyntaxKind => SyntaxKind.InvocationExpression;
+        protected override ILanguageFacade Language => VisualBasicFacade.Instance;
+
+        public SecurityPInvokeMethodShouldNotBeCalled() : base(RspecStrings.ResourceManager) { }
+
+        protected override SyntaxNode Expression(InvocationExpressionSyntax invocationExpression) =>
+            invocationExpression.Expression;
+
+        protected override SyntaxToken Identifier(SyntaxNode syntaxNode) =>
+            ((IdentifierNameSyntax)syntaxNode).Identifier;
+
+        protected override bool IsImportFromInteropDll(ISymbol symbol) =>
+            base.IsImportFromInteropDll(symbol)
+            || (symbol.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax() is DeclareStatementSyntax declaration
+                && IsInterop(declaration.LibraryName?.GetStringValue()));
+
+        protected override string GetMethodName(ISymbol symbol) =>
+            symbol.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax() is DeclareStatementSyntax declaration
+            && declaration.AliasName != null
+                ? declaration.AliasName.GetStringValue()
+                : symbol.Name;
+
+        protected override IMethodSymbol MethodSymbolForInvalidInvocation(SyntaxNode syntaxNode, SemanticModel semanticModel) =>
+            semanticModel.GetSymbolInfo(syntaxNode).Symbol is IMethodSymbol methodSymbol
+            && InvalidMethods.Any(x => x.Equals(GetMethodName(methodSymbol), StringComparison.OrdinalIgnoreCase))
```

Comment:
`GetMethodName` is expensive and it's currently inside loop. Let's extract it to `&& GetMethodName(..) is {} methodName`.

Like that, it would be better to swap the operands to `methodName.Equals(x, ...)`